pkg/reconciler/cluster/cluster_controller_start.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/spf13/pflag"
+)
+
+func DefaultOptions() *Options {
+	return &Options{
+		HeartbeatThreshold: time.Minute,
+	}
+}
+
+func BindOptions(o *Options, fs *pflag.FlagSet) *Options {
+	fs.DurationVar(&o.HeartbeatThreshold, "cluster-heartbeat-threshold", o.HeartbeatThreshold, "Amount of time to wait for a successful heartbeat before marking the cluster as not ready")
+	return o
+}
+
+type Options struct {
+	HeartbeatThreshold time.Duration
+}
+
+func (o *Options) Validate() error {
```

Comment:
[]error as return type

Reply:
All the others return `error` -- I think if we decide to surface multiple errors we should do so using `multierr`.